skimage/io/tests/test_io.py

```diff
@@ -52,3 +53,39 @@ def test_imread_http_url(httpserver):
     # by extension
     image = io.imread(httpserver.url + '/test.jpg' + '?' + 's' * 266)
     assert image.shape == (1, 1)
+
+
+def _named_tempfile_func(error_class):
+    """Create a mock function for NamedTemporaryFile that always raises.
+
+    Parameters
+    ----------
+    error_class : exception class
+        The error that should be raised when asking for a NamedTemporaryFile.
+
+    Returns
+    -------
+    named_temp_file : callable
+        A function that always raises the desired error.
```

Comment:
I would put a list here, shows what kinds of errors we're looking for (which I'm assuming to be `FileNotFoundError`, `FileExistsError`, `PermissionError`, `BaseException`). Is that right @jni? :)